### ISSUE TITLE
Preselect and disable already added variant combinations

### ISSUE THREAD

**Issue by bjarnef** [TeaCommerce/Tea-Commerce-for-Umbraco#47]
When a variant combination already is added, it would make since to preselect it and disable those buttons (or remove them from the options).

Furthermore, based on the starterkit when choosing ```Color: Black, Green``` they are submitted, but you get an error:

```
Some of your variants does not have the same variant types
```

![image](https://cloud.githubusercontent.com/assets/2919859/13444050/53481e98-e004-11e5-96a0-eb095989c6d7.png)

Maybe the "Add variant options" button should be disabled in that case? but I am not sure how I will work when you only have a single variant group like ```Color``` - in that case it should be okay to add variants only from one variant group.

I also thought about if you can configurate the product to use different variant groups? e.g. if you sell a shoe you might want to use variant groups for ```Color``` and ```Shoe Size``` .. but for a T-shirt ```Color``` and ```Size``` ...
so often you want to add variants based on the product type.

Finally it would also be great to demonstrate in demo site, that if you e.g. select ```Color:Black``` it is available in ```Size:S,M,L``` .. but for ```Color:Green``` it is only available in ```Size:S,M``` on frontend (just for demonstration purpose)

**Comment by runegronkjaer** [comment]
All of what you mention here is actually something we either chose by design or it already does it.

1. "Some of your variants does not have the same variant types"
We have chosen to do it like that to give you guys the most possible ways to use the system. There are actually some products out there that have croocked variant groups. You can have that.

2. Also if you want to ensure that ALL variant groups are chosen by the editor you should tick the "Force editor to choose all variant groups" on the variant editor data type. That will disable the add button until at least one of each variant group have been selected.

3. If you sell t-shirts and shoes you can make different variant editor data types and add them to different product document types. Each variant editor data type can have different variant group filters meaning one product could have colors and sizes and others have colors and "Shoe size".

4. The demo site already demonstrates options being disabled after choosing a color. If a size is not in the color it will be disabled. You guys can just change the javascript to remove the ones that can't be chosen if that's the best choise of design for your shop.

/Rune

**Comment by bjarnef** [comment]
Hi Rune

1. Okay, I haven't really tried configuration the "Product" futher, but I guess I could create child document types and composition document types to configurate different product types with different variant groups if you have a store with very different product types. It was also answer for 3.

2. Okay, didn't notice the "Force editor to choose all variant groups" property. However with this checked it only has diabled the add button until one value from one of the groups has been chosen - not both. Any reason why not require one value from each variant group before enable the button as you get an error anyway when only choosing one value (when using multiple variant groups) - see previously posted image.
Futhermore when selection af Color (e.g. Green) "- Select all variant types -" change to "- Deselect all variant types -"  ... it doesn't happen when selection just a Size (e.g. Medium).

4. Ahh okay, I see it now. The contrast between enabled and disabled text i the dropdown options is just low (at least in Chrome). Maybe change to a lighter grey text color for disbled options.

**Comment by runegronkjaer** [comment]
2. I still believe the  "Force editor to choose all variant groups" works:
![not all variant groups selected](https://cloud.githubusercontent.com/assets/3272628/13488455/9d9e56b0-e121-11e5-8019-03186fe4a787.PNG)
I do see that there's a small bug with "- Select all variant types -". I will create a new issue for that.

**Comment by bjarnef** [comment]
Okay, it seems to work now when I check it.

Just when I tried yesterday from my home PC on the demo site, the "Add variant options" button was enabled when I just selected one of the variant groups - also with "Force editor to choose all variant groups" enabled.

Therefore I was able to submit the variants for ```Color: Black, Green``` without a Size selected - I got the "Some of your variants does not have the same variant types" though.

But it seems to work fine now :)
